packages/reaction-core/client/helpers/globals.js

```diff
@@ -72,48 +72,104 @@ this.getProductsByTag = function (tag) {
 };
 
 /**
- * maybeDeleteProduct
- * @summary confirm product deletion, delete, and alert
- * @todo - refactor this back into templates. this is bad.
- * @param {Object} product - product Object
- * @return {Object} - returns nothing, and alerts,happen here
+ * publishProduct
+ * @summary product publishing and alert
+ * @param {Object} productOrArray - product Object
+ * @returns {Object} - returns nothing, and alerts, happen here
  */
-this.maybeDeleteProduct = function (product) {
-  let productIds;
-  let title;
-  let confirmTitle = "Delete this product?";
-
-  if (_.isArray(product)) {
-    if (product.length === 1) {
-      title = product[0].title || "the product";
-      productIds = [product[0]._id];
-    } else {
-      title = "the selected products";
-      confirmTitle = "Delete selected products?";
+publishProduct = function (productOrArray) {
+  let products = _.isArray(productOrArray) === false ? [productOrArray] : productOrArray;
+  for (let product of products) {
+    Meteor.call("products/publishProduct", product._id, function (error, result) {
+      if (error) {
+        Alerts.add(error, "danger", {
+          placement: "productGridItem",
+          id: self._id
+        });
+        return {};
+      }
+      if (result === true) {
+        return Alerts.add(product.title + " " + i18n.t("productDetail.publishProductVisible"), "success", {
+          placement: "productGridItem",
+          id: product._id,
+          autoHide: true,
+          dismissable: false
+        });
+      } else {
+        return Alerts.add(product.title + " " + i18n.t("productDetail.publishProductHidden"), "warning", {
+          placement: "productGridItem",
+          id: product._id,
+          autoHide: true,
+          dismissable: false
+        });
+      }
+    });
+  }
+};
 
-      productIds = _.map(product, (item) => {
-        return item._id;
+/**
+ * cloneProduct
+ * @summary product cloning and alert
+ * @param {Object} productOrArray - product Object
+ * @param {Boolean} redirect - shoud we do redirect after cloning?
+ * @returns {Object} - returns nothing, and alerts, happen here
+ */
+cloneProduct = function (productOrArray, redirect) {
+  let products = _.isArray(productOrArray) === false ? [productOrArray] : productOrArray;
+  return Meteor.call("products/cloneProduct", products, function (error) {
+    if (error) {
+      throw new Meteor.Error("error cloning product", error);
+    }
+    for (let product of products) {
+      Alerts.add(i18n.t("productDetail.clonedAlert") + " " + product.title, "success", {
+        placement: "productGridItem",
+        id: product._id,
+        autoHide: true,
+        dismissable: false
+      });
+    }
+    if (redirect) {
+      Router.go("product", {
+        _id: productId
```

Comment:
As far as I see it is implied if there should be a redirect: It should only happen, if there was a single product passed to this function, not if it's an array. So I would suggest the following:

```js
/**
 * cloneProduct
 * @summary product cloning and alert
 * @param {Object} productOrArray - product Object
 * @returns {Object} - returns nothing, and alerts, happen here
 */
cloneProduct = function (productOrArray) {
  const paramIsProduct = ! _.isArray(productOrArray);
  const products = paramIsProduct ? [productOrArray] : productOrArray;

  return Meteor.call("products/cloneProduct", products, function (error) {
    if (error) {
      throw new Meteor.Error("error cloning product", error);
    }
    for (let product of products) {
      Alerts.add(i18n.t("productDetail.clonedAlert") + " " + product.title, "success", {
        placement: "productGridItem",
        id: product._id,
        autoHide: true,
        dismissable: false
      });
    }

    if (paramIsProduct) {
      Router.go("product", {
        _id: productOrArray._id
      });
    }
  });
};
```